explorer/ui/src/containers/VestingContainer.ts

```diff
@@ -0,0 +1,115 @@
+import ErrorContainer from './ErrorContainer';
+import { CasperService } from 'casperlabs-sdk';
+import { StateQuery } from 'casperlabs-grpc/io/casperlabs/node/api/casper_pb';
+import { observable } from 'mobx';
+
+
+export class VestingContainer {
+  @observable vestingDetails: VestingDetail | null = null;
+
+  constructor(
+    private errors: ErrorContainer,
+    private casperService: CasperService
+  ) {
+  }
+
+  /** Call whenever the page switches to a new vesting contract. */
+  async init(hash: string, showLoading: boolean = false) {
+    // show loading
+    if (showLoading) {
+      this.vestingDetails = null;
+    }
+    this.vestingDetails = await this.getVestingDetails(hash);
+  }
+
+
+  private async getVestingDetails(keyBase16: string) {
+    let res = new VestingDetail();
+    let paths = Object.keys(res);
+    let stateQueries = paths.map(p => {
+      let s = new StateQuery();
+      s.setKeyBase16(keyBase16);
+      s.setKeyVariant(StateQuery.KeyVariant.HASH);
+      s.setPathSegmentsList([p]);
+      return s;
+    });
+    let lastFinalizedBlockInfo = await this.casperService.getLastFinalizedBlockInfo();
+    let values = await this.casperService.batchGetBlockState(lastFinalizedBlockInfo.getSummary()!.getBlockHash_asU8(), stateQueries);
+    for (let i = 0; i < values.length; i++) {
+      (res as any)[paths[i]] = Number(values[i].getBigInt()!.getValue());
+    }
+    return res;
+  }
+}
+
+export class VestingDetail {
+  cliff_timestamp: number;
+  cliff_amount: number;
+  total_amount: number;
+  released_amount: number;
+  on_pause_duration: number;
+  last_pause_timestamp: number;
+  drip_duration: number;
+  drip_amount: number;
+  admin_release_duration: number;
+
+  // Todo fetch from global state storage once the parsing bug is fixed.
+  get admin_account(): string {
+    return "ad1ce8c63f6439c12a6c57f8d797e2a1ea7af76ccdcc08b83baa5f84ffc180f1";
+  }
+
+  // Todo fetch from global state storage once the parsing bug is fixed.
+  get recipient_account(): string {
+    return "400ceb75b8ad14a395edd03a285cc2de745cc61bef22e5a8e214a9783505409c";
+  }
+
+  // Todo fetch from global state storage once the parsing bug is fixed.
+  get is_paused(): boolean{
+    return true;
+  }
+
+  // check whether the contract is releasable by admin account
+  get is_releasable(): boolean{
+    if (!this.is_paused) {
+      return false;
+    }
+    let since_last_pause = Date.now() - this.last_pause_timestamp;
+    if( since_last_pause < this.admin_release_duration ){
```

Comment:
```suggestion
    if (since_last_pause < this.admin_release_duration) {
```
I thought there was some auto-formatter.